encoderfile/tests/test_http.rs

```diff
@@ -0,0 +1,121 @@
+macro_rules! test_router_mod {
```

Comment:
If you came up with tests that apply only to, say, embedding, would it be easy to add this to the macro-generated modules?